packages/wp-now/src/download.ts

```diff
@@ -55,18 +74,34 @@ async function downloadFileAndUnzip({
 				}
 			})
 			.promise();
+		return { downloaded: true, statusCode };
 	} catch (err) {
 		console.error(`Error downloading or unzipping ${itemName}:`, err);
 	}
+	return { downloaded: false, statusCode };
 }
 
-export async function downloadWordPress(fileName = 'latest') {
-	await downloadFileAndUnzip({
-		url: WP_DOWNLOAD_URL,
-		destinationFolder: path.join(WORDPRESS_VERSIONS_PATH, fileName),
-		checkFinalPath: path.join(WORDPRESS_VERSIONS_PATH, fileName),
-		itemName: `WordPress ${fileName}`,
+export async function downloadWordPress(
+	wordPressVersion = DEFAULT_WORDPRESS_VERSION
+) {
+	const finalFolder = path.join(WORDPRESS_VERSIONS_PATH, wordPressVersion);
+	const tempFolder = os.tmpdir();
+	const { downloaded, statusCode } = await downloadFileAndUnzip({
+		url: getWordPressVersionUrl(wordPressVersion),
+		destinationFolder: tempFolder,
+		checkFinalPath: finalFolder,
+		itemName: `WordPress ${wordPressVersion}`,
 	});
+	console.log('downloaded', downloaded);
+	if (downloaded) {
+		fs.ensureDirSync(path.dirname(finalFolder));
+		fs.renameSync(path.join(tempFolder, 'wordpress'), finalFolder);
+	} else if (404 === statusCode) {
+		console.log(
+			`WordPress ${wordPressVersion} not found. Check https://wordpress.org/download/releases/ for available versions.`
+		);
+		process.exit(1);
+	}
```

Comment:
WordPress auto-updated on me 🙃

Should we prevent the behavior? We'll want to [set a constant or add a filter](https://wordpress.org/documentation/article/configuring-automatic-background-updates/).

Reply:
I added this functionality.
Now for WP versions different than `latest` we will disable the auto update core:

https://github.com/WordPress/wordpress-playground/pull/252/commits/48822ea858064e191dc558fd244ed2522f3f92a4

| **Before** | **After** |
|---|---|
| <img width="1074" alt="Screenshot 2023-05-09 at 15 01 37" src="https://github.com/WordPress/wordpress-playground/assets/779993/9784d981-8e1e-4ffb-a425-a9ca05b3eb8d"> |<img width="1074" alt="Screenshot 2023-05-09 at 15 00 32" src="https://github.com/WordPress/wordpress-playground/assets/779993/dd68fce3-ae83-4dd6-a1c8-75c7d9d96b35"> |


I had to use the `define( 'WP_AUTO_UPDATE_CORE', false );` because the filters didn't work.
I tried creating a mu-plugin with:

```php
add_filter( 'auto_update_core', '__return_false' );
add_filter( 'automatic_updater_disabled', '__return_false' );
```